Source/Urho3D/Navigation/NavigationMesh.cpp

```diff
@@ -890,7 +897,7 @@ void NavigationMesh::GetTileGeometry(NavBuildData* build, Vector<NavigationGeome
                 NavArea* area = static_cast<NavArea*>(geometryList[i].component_);
                 NavAreaStub stub;
                 //\todo is there an alternative to casting down? Attributes restricts area ID/type to being an unsigned int

```

Comment:
IMHO, you can remove this TODO comment or at least remove the reference to area "type" in the comment.

As I understand it, in theory there is nothing prevent us from having an ```unsigned char``` type attribute. Currently it does not compile if you try to do so because the attribute macros cannot find templates specialized for ```unsigned char``` and also currently Variant class does not have overload for this data type. But all these can be easily added if we really want to. In practice, however, I don't think it would make much performance difference with what you have done now.

If it really bothering you then perhaps you should change the data type for ```NavArea::areaID_``` private member variable to ```unsigned char``` and make the necessary adjustment to its getter/setter. So, the casting only happens once in the setter. In other words, the "property" is ```unsigned char``` although the "attribute" is ```unsigned```.